cmd/notary/main_test.go

```diff
@@ -369,3 +370,152 @@ func TestConfigFileOverridenByCmdLineFlags(t *testing.T) {
+func TestConfigFileTrustPinning(t *testing.T) {
+	// Set up server that with a self signed cert
+	var err error
+	// add a handler for getting the root
+	m := &recordingMetaStore{MemStorage: *storage.NewMemStorage()}
+	s := httptest.NewUnstartedServer(setupServerHandler(m))
+	s.TLS, err = tlsconfig.Server(tlsconfig.Options{
+		CertFile:   "../../fixtures/notary-server.crt",
+		KeyFile:    "../../fixtures/notary-server.key",
+		CAFile:     "../../fixtures/root-ca.crt",
+		ClientAuth: tls.RequireAndVerifyClientCert,
+	})
+	require.NoError(t, err)
+	s.StartTLS()
+	defer s.Close()
+
+	commander := &notaryCommander{
+		getRetriever: func() passphrase.Retriever { return passphrase.ConstantRetriever("pass") },
+	}
+
+	tempDir := tempDirWithConfig(t, fmt.Sprintf(`{
+		"remote_server": {
+			"url": "%s"
+		},
+		"trust_pinning": {
+		    "tofu": true
+		 }
+	}`, s.URL))
+	defer os.RemoveAll(tempDir)
+	configFile := filepath.Join(tempDir, "config.json")
+
+	// set a config file, so it doesn't check ~/.notary/config.json by default,
+	// and execute a random command so that the flags are parsed
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+
+	cmd := commander.GetCommand()
+	cmd.SetArgs([]string{
+		"-c", configFile, "-d", tempDir, "list", "repo",
+		"--tlscacert", "../../fixtures/root-ca.crt",
+		"--tlscert", filepath.Clean(filepath.Join(cwd, "../../fixtures/notary-server.crt")),
+		"--tlskey", "../../fixtures/notary-server.key"})
+	cmd.SetOutput(new(bytes.Buffer)) // eat the output
+	err = cmd.Execute()
+	require.Error(t, err, "there was no repository, so list should have failed")
+	require.NotContains(t, err.Error(), "TLS", "there was no TLS error though!")
+
+	// Check that tofu was set correctly
+	config, err := commander.parseConfig()
+	require.NoError(t, err)
+	require.Equal(t, false, config.GetBool("trust_pinning.disable_tofu"))
+
+	// validate that we actually managed to connect and attempted to download the root though
+	require.Len(t, m.gotten, 1)
+	require.Equal(t, m.gotten[0], "repo.root")
+
+	tempDir2 := tempDirWithConfig(t, fmt.Sprintf(`{
+		"remote_server": {
+			"url": "%s"
+		},
+		"trust_pinning": {
+		    "tofu": false
+		 }
+	}`, s.URL))
+	defer os.RemoveAll(tempDir)
+	configFile = filepath.Join(tempDir2, "config.json")
+
+	cmd = commander.GetCommand()
+	cmd.SetArgs([]string{
+		"-c", configFile, "-d", tempDir2, "list", "repo2",
+		"--tlscacert", "../../fixtures/root-ca.crt",
+		"--tlscert", filepath.Clean(filepath.Join(cwd, "../../fixtures/notary-server.crt")),
+		"--tlskey", "../../fixtures/notary-server.key"})
+	cmd.SetOutput(new(bytes.Buffer)) // eat the output
+	err = cmd.Execute()
+	require.Error(t, err, "there was no repository, so list should have failed")
+	require.NotContains(t, err.Error(), "TLS", "there was no TLS error though!")
+
+	config, err = commander.parseConfig()
+	require.NoError(t, err)
+	require.Equal(t, false, config.GetBool("trust_pinning.disable_tofu"))
```

Comment:
I'm wondering if it would make sense to pass the config to `getTrustPinning` here rather than to test the viper config itself?  And assert true/false on the `TrustPinConfig` object that comes out?